### PR TITLE
Add tests for the repeated and map enum value fields where the enum value is unknown.

### PR DESCRIPTION
**Pull request by mightyvoice**
Add tests for the repeated and map enum value fields where the enum value is unknown.

### CODE REVIEW

File: protoc_plugin/test/protos/map_enum_value.proto [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+// Test mapped enum value before with unknown values.
+message MapEnumValue {
+  enum NestedEnum {
+    // Unknown values should default to this.
+    UNKNOWN = 0;
+    // A known value to test control.
+    NEW_VALUE = 1;
+  }
+  map<string, NestedEnum> values = 1;
+}
```
Comment (nichite) [inline protoc_plugin/test/protos/map_enum_value.proto:18]: nit: can you add the newline here

**Review by nichite — APPROVED** [review]
LGTM

File: protoc_plugin/test/protos/map_enum_value.proto (hunk 1 above)
Comment (mightyvoice) [reply protoc_plugin/test/protos/map_enum_value.proto]: done

File: protoc_plugin/test/json_test.dart [hunk 2]
```diff
@@ -161,17 +162,30 @@ void main() {
...
     var message = TestAllTypes.fromJson('{"51": [4]}');
-    // 4 is an unknown value.
-    expect(message.repeatedNestedEnum, isEmpty);
+    // 4 is an unknown value, which should default to the enum's default value.
+    expect(message.repeatedNestedEnum, equals([TestAllTypes_NestedEnum.FOO]));
+
+    // 1 (FOO) and 2 (BAR) are known values. All unknowns should fill in to
+    // the default enum value (FOO).
 
     // 1 (FOO) and 2 (BAR) are known values.
```
Comment (nichite) [inline protoc_plugin/test/json_test.dart:171]: Ah, this comment is actually meant to be replaced with the longer above comment.

Comment (mightyvoice) [reply protoc_plugin/test/json_test.dart]: done

**Pull request merged**